src/ppspp_options.erl

```diff
@@ -126,23 +132,25 @@ unpack( <<?PPSPP_MERKLE_HASH_FUNCTION,
                         3 -> sha384;
                         4 -> sha512
                     end,
-    Options = orddict:store(merkle_hash_function, Hash_Function, Options0),
-    unpack(Maybe_Options, Options);
+    %% Options = orddict:store(merkle_hash_function, Hash_Function, Options0),
+    unpack(Maybe_Options, Options#{merkle_hash_function => Hash_Function});
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 unpack( <<?PPSPP_LIVE_SIGNATURE_ALGORITHM,
           Maybe_Algorithm:?BYTE,
-          Maybe_Options/binary >>, Options0) ->
+          Maybe_Options/binary >>, Options) ->
     Algorithm = case Maybe_Algorithm of
                     %% TODO we can parse this yet but need normative algorithm names
```

Comment:
Line 143 is too long:                     %% TODO we can parse this yet but need normative algorithm names.